### ISSUE TITLE
Initial build fails

### ISSUE THREAD

**Issue by mchassy** [ansible/ansible-container#627]
##### ISSUE TYPE
 - Bug Report

##### container.yml 
```
version: "2"
settings:
  conductor_base: centos:7

services:
  aco_service_1:
    from: "centos:7"
    container_name: "aco_service_1"
    expose:
      - 27017
    command: ["sleep", "infinity"]
  aco_service_2:
    from: "centos:7"
    container_name: "aco_service_2"
    ports:
      - 8080:8080
    command: ["sleep", "infinity"]
registries: {}

```

##### main.yml 

```
#!/bin/bash
---
- hosts: aco_service_1
  tasks:
    - name: Create Orson User
      user: name=orson
      regsiter: user_created
    - debug: msg="{{ user_created }}"
- hosts: aco_service_2
  tasks:
    - name: Create Orson User
      user: name=orson
      regsiter: user_created
    - debug: msg="{{ user_created }}"

```

##### OS / ENVIRONMENT
```
$ ansible-container --debug version
Ansible Container, version 0.9.1
Darwin, MMac.local, 16.6.0, Darwin Kernel Version 16.6.0: Fri Apr 14 16:21:16 PDT 2017; root:xnu-3789.60.24~6/RELEASE_X86_64, x86_64
2.7.12 (v2.7.12:d33e0cf91556, Jun 26 2016, 12:10:39)
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] /Users/markchassy/venv2/bin/python
{
  "ContainersPaused": 0,
  "Labels": null,
  "CgroupDriver": "cgroupfs",
  "ContainersRunning": 0,
  "ContainerdCommit": {
    "Expected": "cfb82a876ecc11b5ca0977d1733adbe58599088a",
    "ID": "cfb82a876ecc11b5ca0977d1733adbe58599088a"
  },
  "InitBinary": "docker-init",
  "NGoroutines": 31,
  "Swarm": {
    "ControlAvailable": false,
    "NodeID": "",
    "Error": "",
    "RemoteManagers": null,
    "LocalNodeState": "inactive",
    "NodeAddr": ""
  },
  "LoggingDriver": "json-file",
  "OSType": "linux",
  "HttpProxy": "",
  "Runtimes": {
    "runc": {
      "path": "docker-runc"
    }
  },
  "DriverStatus": [
    [
      "Root Dir",
      "/var/lib/docker/aufs"
    ],
    [
      "Backing Filesystem",
      "extfs"
    ],
    [
      "Dirs",
      "41"
    ],
    [
      "Dirperm1 Supported",
      "true"
    ]
  ],
  "OperatingSystem": "Alpine Linux v3.5",
  "Containers": 1,
  "HttpsProxy": "",
  "BridgeNfIp6tables": true,
  "MemTotal": 4139139072,
  "SecurityOptions": [
    "name=seccomp,profile=default"
  ],
  "Driver": "aufs",
  "IndexServerAddress": "https://index.docker.io/v1/",
  "ClusterStore": "",
  "InitCommit": {
    "Expected": "949e6fa",
    "ID": "949e6fa"
  },
  "Isolation": "",
  "SystemStatus": null,
  "OomKillDisable": true,
  "ClusterAdvertise": "",
  "SystemTime": "2017-06-29T00:30:13.383339061Z",
  "Name": "moby",
  "CPUSet": true,
  "RegistryConfig": {
    "AllowNondistributableArtifactsCIDRs": [],
    "Mirrors": [],
    "IndexConfigs": {
      "docker.io": {
        "Official": true,
        "Name": "docker.io",
        "Secure": true,
        "Mirrors": []
      }
    },
    "AllowNondistributableArtifactsHostnames": [],
    "InsecureRegistryCIDRs": [
      "127.0.0.0/8"
    ]
  },
  "DefaultRuntime": "runc",
  "ContainersStopped": 1,
  "NCPU": 4,
  "NFd": 19,
  "Architecture": "x86_64",
  "KernelMemory": true,
  "CpuCfsQuota": true,
  "Debug": true,
  "ID": "7NOW:FSRW:YHXB:5HAT:TFUV:MHSI:GFLT:6WR2:YJCD:TIFP:UQK2:Z7FA",
  "IPv4Forwarding": true,
  "KernelVersion": "4.9.31-moby",
  "BridgeNfIptables": true,
  "NoProxy": "*.local, 169.254/16",
  "LiveRestoreEnabled": false,
  "ServerVersion": "17.06.0-ce",
  "CpuCfsPeriod": true,
  "ExperimentalBuild": true,
  "MemoryLimit": true,
  "SwapLimit": true,
  "Plugins": {
    "Volume": [
      "local"
    ],
    "Network": [
      "bridge",
      "host",
      "ipvlan",
      "macvlan",
      "null",
      "overlay"
    ],
    "Authorization": null,
    "Log": [
      "awslogs",
      "fluentd",
      "gcplogs",
      "gelf",
      "journald",
      "json-file",
      "logentries",
      "splunk",
      "syslog"
    ]
  },
  "Images": 42,
  "DockerRootDir": "/var/lib/docker",
  "NEventsListener": 1,
  "CPUShares": true,
  "RuncCommit": {
    "Expected": "2d41c047c83e09a6d61d464906feb2a2f3c52aa4",
    "ID": "2d41c047c83e09a6d61d464906feb2a2f3c52aa4"
  }
}
{
  "KernelVersion": "4.9.31-moby",
  "Arch": "amd64",
  "BuildTime": "2017-06-23T21:51:55.152028673+00:00",
  "ApiVersion": "1.30",
  "Version": "17.06.0-ce",
  "MinAPIVersion": "1.12",
  "GitCommit": "02c1d87",
  "Os": "linux",
  "Experimental": true,
  "GoVersion": "go1.8.3"
}
(venv2)

```
##### SUMMARY
First steps to start using ansible-container. 

- In compose.yml
Build a 2 services app. Expose a port internally on one, expose a port externally on another, run sleep infinity on both. Both based on Centos/7. 

- In main.yml
Create a new user (orson) in each container.

##### STEPS TO REPRODUCE
`ansible-container --debug build`

```
ansible-container --debug build
```

##### EXPECTED RESULTS

- Conductor image created
- Both service images created

##### ACTUAL RESULTS

- Nothing created

```
~/github/ac_firststep
$ ansible-container --debug build
2017-06-28T20:19:43.445591 The default type is            [container.config] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/config.py caller_func=_resolve_defaults caller_line=140 config=<class 'ruamel.yaml.comments.CommentedMap'> defaults=<type '_ordereddict.ordereddict'>
2017-06-28T20:19:43.446480 Getting environment variables... [container.config] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/config.py caller_func=_get_environment_variables caller_line=156
2017-06-28T20:19:43.447265 Read environment variables     [container.config] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/config.py caller_func=_get_environment_variables caller_line=161 env_vars={}
2017-06-28T20:19:43.447939 Resolved template variables    [container.config] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/config.py caller_func=_resolve_defaults caller_line=145 template_vars={}
2017-06-28T20:19:43.448589 Parsed config                  [container.config] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/config.py caller_func=set_env caller_line=121 config={"version": "2", "settings": {"conductor_base": "centos:7", "pwd": "/Users/markchassy/github/ac_firststep"}, "services": {"aco_service_1": {"from": "centos:7", "container_name": "aco_service_1", "expose": [27017], "command": ["sleep", "infinity"]}, "aco_service_2": {"from": "centos:7", "container_name": "aco_service_2", "ports": ["8080:8080"], "command": ["sleep", "infinity"]}}, "registries": {}, "defaults": {}}
2017-06-28T20:19:43.449370 Loading engine capabilities    [container.utils.loader] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/utils/loader.py caller_func=load_engine caller_line=14 capabilities=['BUILD', 'RUN'] engine=docker
2017-06-28T20:19:43.486253 Could not find container for conductor [container.docker.engine] all_containers=[] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/docker/engine.py caller_func=get_container_id_for_service caller_line=384 container=u'ac_firststep_conductor'
2017-06-28T20:19:43.501189 Could not find the latest image for service, searching for other tags with same image name [container.docker.engine] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/docker/engine.py caller_func=get_latest_image_for_service caller_line=423 image_name=u'ac_firststep-conductor' service=conductor
2017-06-28T20:19:43.504439 Call: Engine.build_conductor_image [container.docker.engine] args=('/Users/markchassy/github/ac_firststep', 'centos:7') caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/docker/engine.py caller_func=Engine.build_conductor_image caller_line=23 kwargs={'cache': True}
2017-06-28T20:19:43.506973 Using temporary directory      [container.utils.temp] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/utils/temp.py caller_func=__enter__ caller_line=17 path=/var/folders/fm/6g1xbv4938g1ld3rd2l_dlyh0000gn/T/tmpKa5F8R
2017-06-28T20:19:43.507548 Building Docker Engine context... [container.docker.engine] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/docker/engine.py caller_func=build_conductor_image caller_line=649
2017-06-28T20:19:43.550529 Rendered Jinja Template:       [container.utils] body=FROM centos:7
ENV ANSIBLE_CONTAINER=1


RUN yum update -y && \
    yum install -y epel-release && \
    yum install -y gcc git python-devel rsync libffi-devel openssl-devel && \
    yum clean all


ADD https://get.docker.com/builds/Linux/x86_64/docker-17.04.0-ce.tgz /tmp/docker.tgz

COPY /contrib/get-pip.py /get-pip.py
RUN python /get-pip.py && \
    mkdir -p /etc/ansible/roles /_ansible/src && \
    cd /usr/local/bin && \
    tar -xz --strip-components=1 -f /tmp/docker.tgz

# The COPY here will break cache if the version of conductor changed
COPY /container-src /_ansible/container
RUN cd /_ansible && \
    pip install -r container/conductor-build/conductor-requirements.txt && \
    PYTHONPATH=. LC_ALL="en_US.UTF-8" python container/conductor-build/setup.py develop -v && \
    ansible-galaxy install -p /etc/ansible/roles -r container/conductor-build/conductor-requirements.yml

# The COPY here will break cache if the requirements or ansible.cfg has changed
COPY /build-src /_ansible/build
RUN ( test -f /_ansible/build/ansible-requirements.txt && pip install --no-cache-dir -r /_ansible/build/ansible-requirements.txt || true ) && \
    ( test -f /_ansible/build/requirements.yml && ansible-galaxy install -p /etc/ansible/roles -r /_ansible/build/requirements.yml || true ) && \
    ( test -f /_ansible/build/ansible.cfg && cp /_ansible/build/ansible.cfg /etc/ansible/ansible.cfg || true)

VOLUME /usr


 caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/utils/__init__.py caller_func=jinja_render_to_temp caller_line=81
2017-06-28T20:19:43.551546 Context manifest:              [container.docker.engine] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/docker/engine.py caller_func=build_conductor_image caller_line=705
2017-06-28T20:19:43.551693 tarball item: build-src/requirements.yml (297 bytes) [container.docker.engine] bytes=297 file=build-src/requirements.yml
2017-06-28T20:19:43.551872 tarball item: build-src/.touch (0 bytes) [container.docker.engine] bytes=0 file=build-src/.touch
2017-06-28T20:19:43.552045 tarball item: contrib/get-pip.py (1595409 bytes) [container.docker.engine] bytes=1595409 file=contrib/get-pip.py
2017-06-28T20:19:43.552215 tarball item: container-src (0 bytes) [container.docker.engine] bytes=0L file=container-src
2017-06-28T20:19:43.552384 tarball item: container-src/__init__.py (947 bytes) [container.docker.engine] bytes=947 file=container-src/__init__.py
2017-06-28T20:19:43.552530 tarball item: container-src/__init__.pyc (1466 bytes) [container.docker.engine] bytes=1466 file=container-src/__init__.pyc
2017-06-28T20:19:43.552675 tarball item: container-src/cli.py (19540 bytes) [container.docker.engine] bytes=19540 file=container-src/cli.py
2017-06-28T20:19:43.552820 tarball item: container-src/cli.pyc (15911 bytes) [container.docker.engine] bytes=15911 file=container-src/cli.pyc
2017-06-28T20:19:43.552953 tarball item: container-src/config.py (14021 bytes) [container.docker.engine] bytes=14021 file=container-src/config.py
2017-06-28T20:19:43.553107 tarball item: container-src/config.pyc (13055 bytes) [container.docker.engine] bytes=13055 file=container-src/config.pyc
2017-06-28T20:19:43.553249 tarball item: container-src/core.py (37981 bytes) [container.docker.engine] bytes=37981 file=container-src/core.py
2017-06-28T20:19:43.553390 tarball item: container-src/core.pyc (27523 bytes) [container.docker.engine] bytes=27523 file=container-src/core.pyc
2017-06-28T20:19:43.553531 tarball item: container-src/docker (0 bytes) [container.docker.engine] bytes=0L file=container-src/docker
2017-06-28T20:19:43.553671 tarball item: container-src/docker/__init__.py (205 bytes) [container.docker.engine] bytes=205 file=container-src/docker/__init__.py
2017-06-28T20:19:43.553817 tarball item: container-src/docker/__init__.pyc (342 bytes) [container.docker.engine] bytes=342 file=container-src/docker/__init__.pyc
2017-06-28T20:19:43.553939 tarball item: container-src/docker/deploy.py (248 bytes) [container.docker.engine] bytes=248 file=container-src/docker/deploy.py
2017-06-28T20:19:43.554041 tarball item: container-src/docker/deploy.pyc (340 bytes) [container.docker.engine] bytes=340 file=container-src/docker/deploy.pyc
2017-06-28T20:19:43.554143 tarball item: container-src/docker/engine.py (34750 bytes) [container.docker.engine] bytes=34750 file=container-src/docker/engine.py
2017-06-28T20:19:43.554244 tarball item: container-src/docker/engine.pyc (29885 bytes) [container.docker.engine] bytes=29885 file=container-src/docker/engine.pyc
2017-06-28T20:19:43.554346 tarball item: container-src/docker/files (0 bytes) [container.docker.engine] bytes=0L file=container-src/docker/files
2017-06-28T20:19:43.554448 tarball item: container-src/docker/files/conductor-requirements.txt (294 bytes) [container.docker.engine] bytes=294 file=container-src/docker/files/conductor-requirements.txt
2017-06-28T20:19:43.554550 tarball item: container-src/docker/files/conductor-requirements.yml (60 bytes) [container.docker.engine] bytes=60 file=container-src/docker/files/conductor-requirements.yml
2017-06-28T20:19:43.554651 tarball item: container-src/docker/files/get-pip.py (1595409 bytes) [container.docker.engine] bytes=1595409 file=container-src/docker/files/get-pip.py
2017-06-28T20:19:43.554753 tarball item: container-src/docker/files/get-pip.pyc (1593478 bytes) [container.docker.engine] bytes=1593478 file=container-src/docker/files/get-pip.pyc
2017-06-28T20:19:43.554855 tarball item: container-src/docker/files/setup.py (3215 bytes) [container.docker.engine] bytes=3215 file=container-src/docker/files/setup.py
2017-06-28T20:19:43.554956 tarball item: container-src/docker/files/setup.pyc (4110 bytes) [container.docker.engine] bytes=4110 file=container-src/docker/files/setup.pyc
2017-06-28T20:19:43.555057 tarball item: container-src/docker/importer.py (23310 bytes) [container.docker.engine] bytes=23310 file=container-src/docker/importer.py
2017-06-28T20:19:43.555158 tarball item: container-src/docker/importer.pyc (20113 bytes) [container.docker.engine] bytes=20113 file=container-src/docker/importer.pyc
2017-06-28T20:19:43.555258 tarball item: container-src/docker/templates (0 bytes) [container.docker.engine] bytes=0L file=container-src/docker/templates
2017-06-28T20:19:43.555359 tarball item: container-src/docker/templates/conductor-dockerfile.j2 (2901 bytes) [container.docker.engine] bytes=2901 file=container-src/docker/templates/conductor-dockerfile.j2
2017-06-28T20:19:43.555460 tarball item: container-src/engine.py (6074 bytes) [container.docker.engine] bytes=6074 file=container-src/engine.py
2017-06-28T20:19:43.555561 tarball item: container-src/engine.pyc (10184 bytes) [container.docker.engine] bytes=10184 file=container-src/engine.pyc
2017-06-28T20:19:43.555662 tarball item: container-src/exceptions.py (1991 bytes) [container.docker.engine] bytes=1991 file=container-src/exceptions.py
2017-06-28T20:19:43.555762 tarball item: container-src/exceptions.pyc (5585 bytes) [container.docker.engine] bytes=5585 file=container-src/exceptions.pyc
2017-06-28T20:19:43.555862 tarball item: container-src/k8s (0 bytes) [container.docker.engine] bytes=0L file=container-src/k8s
2017-06-28T20:19:43.555962 tarball item: container-src/k8s/__init__.py (196 bytes) [container.docker.engine] bytes=196 file=container-src/k8s/__init__.py
2017-06-28T20:19:43.556063 tarball item: container-src/k8s/__init__.pyc (339 bytes) [container.docker.engine] bytes=339 file=container-src/k8s/__init__.pyc
2017-06-28T20:19:43.556163 tarball item: container-src/k8s/base_deploy.py (27642 bytes) [container.docker.engine] bytes=27642 file=container-src/k8s/base_deploy.py
2017-06-28T20:19:43.556263 tarball item: container-src/k8s/base_deploy.pyc (20341 bytes) [container.docker.engine] bytes=20341 file=container-src/k8s/base_deploy.pyc
2017-06-28T20:19:43.556364 tarball item: container-src/k8s/base_engine.py (5164 bytes) [container.docker.engine] bytes=5164 file=container-src/k8s/base_engine.py
2017-06-28T20:19:43.556465 tarball item: container-src/k8s/base_engine.pyc (5633 bytes) [container.docker.engine] bytes=5633 file=container-src/k8s/base_engine.pyc
2017-06-28T20:19:43.556566 tarball item: container-src/k8s/deploy.py (1720 bytes) [container.docker.engine] bytes=1720 file=container-src/k8s/deploy.py
2017-06-28T20:19:43.556667 tarball item: container-src/k8s/deploy.pyc (2036 bytes) [container.docker.engine] bytes=2036 file=container-src/k8s/deploy.pyc
2017-06-28T20:19:43.556768 tarball item: container-src/k8s/engine.py (2389 bytes) [container.docker.engine] bytes=2389 file=container-src/k8s/engine.py
2017-06-28T20:19:43.556868 tarball item: container-src/k8s/engine.pyc (1968 bytes) [container.docker.engine] bytes=1968 file=container-src/k8s/engine.pyc
2017-06-28T20:19:43.556968 tarball item: container-src/openshift (0 bytes) [container.docker.engine] bytes=0L file=container-src/openshift
2017-06-28T20:19:43.557068 tarball item: container-src/openshift/__init__.py (196 bytes) [container.docker.engine] bytes=196 file=container-src/openshift/__init__.py
2017-06-28T20:19:43.557168 tarball item: container-src/openshift/__init__.pyc (345 bytes) [container.docker.engine] bytes=345 file=container-src/openshift/__init__.pyc
2017-06-28T20:19:43.557267 tarball item: container-src/openshift/deploy.py (6203 bytes) [container.docker.engine] bytes=6203 file=container-src/openshift/deploy.py
2017-06-28T20:19:43.557367 tarball item: container-src/openshift/deploy.pyc (5319 bytes) [container.docker.engine] bytes=5319 file=container-src/openshift/deploy.pyc
2017-06-28T20:19:43.557467 tarball item: container-src/openshift/engine.py (2151 bytes) [container.docker.engine] bytes=2151 file=container-src/openshift/engine.py
2017-06-28T20:19:43.557568 tarball item: container-src/openshift/engine.pyc (2696 bytes) [container.docker.engine] bytes=2696 file=container-src/openshift/engine.pyc
2017-06-28T20:19:43.557668 tarball item: container-src/templates (0 bytes) [container.docker.engine] bytes=0L file=container-src/templates
2017-06-28T20:19:43.557768 tarball item: container-src/templates/ansible-container-inventory.py (1230 bytes) [container.docker.engine] bytes=1230 file=container-src/templates/ansible-container-inventory.py
2017-06-28T20:19:43.557871 tarball item: container-src/templates/ansible-container-inventory.pyc (2107 bytes) [container.docker.engine] bytes=2107 file=container-src/templates/ansible-container-inventory.pyc
2017-06-28T20:19:43.557978 tarball item: container-src/templates/ansible-dockerfile.j2 (1194 bytes) [container.docker.engine] bytes=1194 file=container-src/templates/ansible-dockerfile.j2
2017-06-28T20:19:43.558088 tarball item: container-src/templates/ansible.cfg (41 bytes) [container.docker.engine] bytes=41 file=container-src/templates/ansible.cfg
2017-06-28T20:19:43.558199 tarball item: container-src/templates/build-docker-compose.j2.yml (1635 bytes) [container.docker.engine] bytes=1635 file=container-src/templates/build-docker-compose.j2.yml
2017-06-28T20:19:43.558311 tarball item: container-src/templates/builder.sh (620 bytes) [container.docker.engine] bytes=620 file=container-src/templates/builder.sh
2017-06-28T20:19:43.558422 tarball item: container-src/templates/compose_versioned.j2.yml (237 bytes) [container.docker.engine] bytes=237 file=container-src/templates/compose_versioned.j2.yml
2017-06-28T20:19:43.558540 tarball item: container-src/templates/hosts.j2 (90 bytes) [container.docker.engine] bytes=90 file=container-src/templates/hosts.j2
2017-06-28T20:19:43.558657 tarball item: container-src/templates/init (0 bytes) [container.docker.engine] bytes=0L file=container-src/templates/init
2017-06-28T20:19:43.558774 tarball item: container-src/templates/init/ansible-requirements.j2.txt (130 bytes) [container.docker.engine] bytes=130 file=container-src/templates/init/ansible-requirements.j2.txt
2017-06-28T20:19:43.558874 tarball item: container-src/templates/init/ansible.j2.cfg (145 bytes) [container.docker.engine] bytes=145 file=container-src/templates/init/ansible.j2.cfg
2017-06-28T20:19:43.558983 tarball item: container-src/templates/init/container.j2.yml (2116 bytes) [container.docker.engine] bytes=2116 file=container-src/templates/init/container.j2.yml
2017-06-28T20:19:43.559084 tarball item: container-src/templates/init/meta.j2.yml (1231 bytes) [container.docker.engine] bytes=1231 file=container-src/templates/init/meta.j2.yml
2017-06-28T20:19:43.559185 tarball item: container-src/templates/init/requirements.j2.yml (298 bytes) [container.docker.engine] bytes=298 file=container-src/templates/init/requirements.j2.yml
2017-06-28T20:19:43.559285 tarball item: container-src/templates/install-docker-compose.j2.yml (712 bytes) [container.docker.engine] bytes=712 file=container-src/templates/install-docker-compose.j2.yml
2017-06-28T20:19:43.559386 tarball item: container-src/templates/listhosts-docker-compose.j2.yml (1640 bytes) [container.docker.engine] bytes=1640 file=container-src/templates/listhosts-docker-compose.j2.yml
2017-06-28T20:19:43.559486 tarball item: container-src/templates/restart-docker-compose.j2.yml (75 bytes) [container.docker.engine] bytes=75 file=container-src/templates/restart-docker-compose.j2.yml
2017-06-28T20:19:43.559587 tarball item: container-src/templates/role (0 bytes) [container.docker.engine] bytes=0L file=container-src/templates/role
2017-06-28T20:19:43.559688 tarball item: container-src/templates/role/.travis.j2.yml (540 bytes) [container.docker.engine] bytes=540 file=container-src/templates/role/.travis.j2.yml
2017-06-28T20:19:43.559789 tarball item: container-src/templates/role/defaults (0 bytes) [container.docker.engine] bytes=0L file=container-src/templates/role/defaults
2017-06-28T20:19:43.559889 tarball item: container-src/templates/role/defaults/main.j2.yml (55 bytes) [container.docker.engine] bytes=55 file=container-src/templates/role/defaults/main.j2.yml
2017-06-28T20:19:43.559989 tarball item: container-src/templates/role/meta (0 bytes) [container.docker.engine] bytes=0L file=container-src/templates/role/meta
2017-06-28T20:19:43.560090 tarball item: container-src/templates/role/meta/main.j2.yml (3220 bytes) [container.docker.engine] bytes=3220 file=container-src/templates/role/meta/main.j2.yml
2017-06-28T20:19:43.560190 tarball item: container-src/templates/role/README.j2.md (1287 bytes) [container.docker.engine] bytes=1287 file=container-src/templates/role/README.j2.md
2017-06-28T20:19:43.560290 tarball item: container-src/templates/role/test (0 bytes) [container.docker.engine] bytes=0L file=container-src/templates/role/test
2017-06-28T20:19:43.560390 tarball item: container-src/templates/role/test/test.j2.yml (81 bytes) [container.docker.engine] bytes=81 file=container-src/templates/role/test/test.j2.yml
2017-06-28T20:19:43.560490 tarball item: container-src/templates/run-docker-compose.j2.yml (75 bytes) [container.docker.engine] bytes=75 file=container-src/templates/run-docker-compose.j2.yml
2017-06-28T20:19:43.560599 tarball item: container-src/templates/stop-docker-compose.j2.yml (75 bytes) [container.docker.engine] bytes=75 file=container-src/templates/stop-docker-compose.j2.yml
2017-06-28T20:19:43.560709 tarball item: container-src/templates/wait_on_host.py (2365 bytes) [container.docker.engine] bytes=2365 file=container-src/templates/wait_on_host.py
2017-06-28T20:19:43.560817 tarball item: container-src/templates/wait_on_host.pyc (2513 bytes) [container.docker.engine] bytes=2513 file=container-src/templates/wait_on_host.pyc
2017-06-28T20:19:43.560923 tarball item: container-src/utils (0 bytes) [container.docker.engine] bytes=0L file=container-src/utils
2017-06-28T20:19:43.561030 tarball item: container-src/utils/__init__.py (10744 bytes) [container.docker.engine] bytes=10744 file=container-src/utils/__init__.py
2017-06-28T20:19:43.561137 tarball item: container-src/utils/__init__.pyc (12048 bytes) [container.docker.engine] bytes=12048 file=container-src/utils/__init__.pyc
2017-06-28T20:19:43.561244 tarball item: container-src/utils/_text.py (12304 bytes) [container.docker.engine] bytes=12304 file=container-src/utils/_text.py
2017-06-28T20:19:43.561350 tarball item: container-src/utils/_text.pyc (8841 bytes) [container.docker.engine] bytes=8841 file=container-src/utils/_text.pyc
2017-06-28T20:19:43.561456 tarball item: container-src/utils/galaxy.py (8120 bytes) [container.docker.engine] bytes=8120 file=container-src/utils/galaxy.py
2017-06-28T20:19:43.561562 tarball item: container-src/utils/galaxy.pyc (8013 bytes) [container.docker.engine] bytes=8013 file=container-src/utils/galaxy.pyc
2017-06-28T20:19:43.561675 tarball item: container-src/utils/loader.py (906 bytes) [container.docker.engine] bytes=906 file=container-src/utils/loader.py
2017-06-28T20:19:43.561782 tarball item: container-src/utils/loader.pyc (1200 bytes) [container.docker.engine] bytes=1200 file=container-src/utils/loader.pyc
2017-06-28T20:19:43.561918 tarball item: container-src/utils/logmux.py (1313 bytes) [container.docker.engine] bytes=1313 file=container-src/utils/logmux.py
2017-06-28T20:19:43.562035 tarball item: container-src/utils/logmux.pyc (2694 bytes) [container.docker.engine] bytes=2694 file=container-src/utils/logmux.pyc
2017-06-28T20:19:43.562138 tarball item: container-src/utils/temp.py (698 bytes) [container.docker.engine] bytes=698 file=container-src/utils/temp.py
2017-06-28T20:19:43.562240 tarball item: container-src/utils/temp.pyc (1418 bytes) [container.docker.engine] bytes=1418 file=container-src/utils/temp.pyc
2017-06-28T20:19:43.562342 tarball item: container-src/utils/visibility.py (3329 bytes) [container.docker.engine] bytes=3329 file=container-src/utils/visibility.py
2017-06-28T20:19:43.562443 tarball item: container-src/utils/visibility.pyc (4283 bytes) [container.docker.engine] bytes=4283 file=container-src/utils/visibility.pyc
2017-06-28T20:19:43.562543 tarball item: container-src/conductor-build/setup.py (3215 bytes) [container.docker.engine] bytes=3215 file=container-src/conductor-build/setup.py
2017-06-28T20:19:43.562643 tarball item: container-src/conductor-build/conductor-requirements.txt (294 bytes) [container.docker.engine] bytes=294 file=container-src/conductor-build/conductor-requirements.txt
2017-06-28T20:19:43.562745 tarball item: container-src/conductor-build/conductor-requirements.yml (60 bytes) [container.docker.engine] bytes=60 file=container-src/conductor-build/conductor-requirements.yml
2017-06-28T20:19:43.562846 tarball item: Dockerfile (1417 bytes) [container.docker.engine] bytes=1417 file=Dockerfile
2017-06-28T20:19:43.569594 Starting Docker build of Ansible Container Conductor image (please be patient)... [container.docker.engine] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/docker/engine.py caller_func=build_conductor_image caller_line=713
Step 1/11 : FROM centos:7
 ---> 3bee3060bfc8
Step 2/11 : ENV ANSIBLE_CONTAINER 1
 ---> Using cache
 ---> 81164cabd160
Step 3/11 : RUN yum update -y &&     yum install -y epel-release &&     yum install -y gcc git python-devel rsync libffi-devel openssl-devel &&     yum clean all
 ---> Using cache
 ---> a2366b5a89fe
Step 4/11 : ADD https://get.docker.com/builds/Linux/x86_64/docker-17.04.0-ce.tgz /tmp/docker.tgz

 ---> f4befa161fa6
Removing intermediate container 74d9a8d77d5f
Step 5/11 : COPY /contrib/get-pip.py /get-pip.py
 ---> cc80e6e7ff10
Removing intermediate container 7599f1661592
Step 6/11 : RUN python /get-pip.py &&     mkdir -p /etc/ansible/roles /_ansible/src &&     cd /usr/local/bin &&     tar -xz --strip-components=1 -f /tmp/docker.tgz
 ---> Running in 1cd57e0ea6bb
Collecting pip
  Downloading pip-9.0.1-py2.py3-none-any.whl (1.3MB)
Collecting setuptools
  Downloading setuptools-36.0.1-py2.py3-none-any.whl (476kB)
Collecting wheel
  Downloading wheel-0.29.0-py2.py3-none-any.whl (66kB)
Installing collected packages: pip, setuptools, wheel
Successfully installed pip-9.0.1 setuptools-36.0.1 wheel-0.29.0
tar (child): /tmp/docker.tgz: Cannot read: Is a directory
tar (child): At beginning of tape, quitting now
tar (child): Error is not recoverable: exiting now


gzip: stdin: unexpected end of file
tar: Child returned status 2
tar: Error is not recoverable: exiting now

2017-06-28T20:21:12.588825 Cleaning up temporary directory [container.utils.temp] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/container/utils/temp.py caller_func=__exit__ caller_line=22 path=/var/folders/fm/6g1xbv4938g1ld3rd2l_dlyh0000gn/T/tmpKa5F8R
2017-06-28T20:21:12.590605 Unknown exception Error building conductor image: The command '/bin/sh -c python /get-pip.py &&     mkdir -p /etc/ansible/roles /_ansible/src &&     cd /usr/local/bin &&     tar -xz --strip-components=1 -f /tmp/docker.tgz' returned a non-zero code: 2 [container.cli] caller_file=/Users/markchassy/venv2/lib/python2.7/site-packages/structlog/stdlib.py caller_func=exception caller_line=95
Traceback (most recent call last):
  File "/Users/markchassy/venv2/lib/python2.7/site-packages/container/cli.py", line 268, in __call__
    getattr(core, u'hostcmd_{}'.format(args.subcommand))(**vars(args))
  File "/Users/markchassy/venv2/lib/python2.7/site-packages/container/__init__.py", line 28, in __wrapped__
    return fn(*args, **kwargs)
  File "/Users/markchassy/venv2/lib/python2.7/site-packages/container/core.py", line 150, in hostcmd_build
    cache=kwargs['cache']
  File "/Users/markchassy/venv2/lib/python2.7/site-packages/container/docker/engine.py", line 82, in __wrapped__
    return fn(self, *args, **kwargs)
  File "/Users/markchassy/venv2/lib/python2.7/site-packages/container/__init__.py", line 28, in __wrapped__
    return fn(*args, **kwargs)
  File "/Users/markchassy/venv2/lib/python2.7/site-packages/container/docker/engine.py", line 729, in build_conductor_image
    "Error building conductor image: {0}".format(line_json['errorDetail']['message']))
AnsibleContainerException: Error building conductor image: The command '/bin/sh -c python /get-pip.py &&     mkdir -p /etc/ansible/roles /_ansible/src &&     cd /usr/local/bin &&     tar -xz --strip-components=1 -f /tmp/docker.tgz' returned a non-zero code: 2
(venv2)

markchassy@MMac on Wed Jun 28 at 08:21 pm
~/github/ac_firststep
```

##### ADDITIONAL COMMENTS
I have created a complete install of my product with custom tomee and mongo containers by using base images which have ansible already installed and running the playbook from the Dockerfile. This already works and I love it (and so do my customers). I like the idea of ansible-container even better, but I must say that the documentation, especially compared to the excellent documentation for standard ansible is sorely lacking. I was up and running with ansible in day, and in 2 weeks, I had mastered many of the concepts. I have spent over a week with ansible-container, and I am pretty much stuck.

**Comment by mchassy** [comment]
I think that I have isolated the problem.
Using the debug output above, I recreated a Dockerfile walked through the steps.
This line seems to be the problem.
`Step 4/11 : ADD https://get.docker.com/builds/Linux/x86_64/docker-17.04.0-ce.tgz /tmp/docker.tgz`
If you stop here, and run the container, docker.tgz has already been untarred to a directory called docker.tgz.

Not sure, why that is happening, could be because of the base image being used?

**Comment by mchassy** [comment]
https://github.com/moby/moby/issues/2369

**Comment by chouseknecht** [comment]
@mchassy 

Thanks for using Ansible Container, and for taking the time to open an issue and share your feedback. 

This issue was pointed out earlier and addressed on issue #592. PR #603 provided a workaround, and was merged.

To get the change, you will need to run from source. See our [Running from source](http://docs.ansible.com/ansible-container/installation.html#running-from-source) guide for assistance.

Closing this, as I believe it has been addressed. However, if you still run into trouble, please feel free to reopen.

**Comment by mchassy** [comment]
Thanks Chris.
After fiddling with some python package versions, I got this working.
My conductor container is building and my service containers are starting to build !!
Really fantastic.
